moj-node/server/middleware/auth.js

```diff
@@ -15,6 +15,19 @@ function createNotification(text) {
   return { text };
 }
 
+function isValidUsername(username) {
+  const pattern = new RegExp(/[A-Z][0-9]{4}[A-Z]{2}/i);
```

Comment:
So I don't forget between now and tomorrow, we should be allowing lowercase and just uppercasing before we use it.

Reply:
We have the `/i` in there to make it case insensitive, like you said we can uppercase before usage, we can possibly move this to a utils function as we have this check in several places

Reply:
Doh, didn't spot that :)